### PR TITLE
Add L7 HTTP method test

### PR DESCRIPTION
**Pull request by nebril**

### CODE REVIEW

File: connectivity/tests/pod.go [hunk 1]
```diff
@@ -108,7 +136,11 @@ func curlEndpoints(ctx context.Context, s check.Scenario, t *check.Test,
 				"X-Very-Secret-Token": "42",
 			})
 			t.NewAction(s, epName, client, ep).Run(func(a *check.Action) {
-				a.ExecInPod(ctx, curl(ep, "-H", "X-Very-Secret-Token: 42"))
+				opts := make([]string, 0, len(curlOpts)+2)
```
Comment (sayboras) [inline connectivity/tests/pod.go:139]: +2 or +1 :thinking:

File: connectivity/manifests/client-egress-l7-http-method.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+---
+# client2 is allowed to contact the echo Pod
+# on port 8080 via POST method. HTTP introspection is enabled for client2.
+# The toFQDNs section relies on DNS introspection being performed by
+# the client-egress-only-dns policy.
+apiVersion: "cilium.io/v2"
+kind: CiliumNetworkPolicy
+metadata:
+  namespace: cilium-test
+  name: client-egress-l7-http-method
```
Comment (sayboras) [inline connectivity/manifests/client-egress-l7-http-method.yaml:10]: nit: test name could be same as policy name for faster reference (e.g. client-egress-l7-method)

**Review by sayboras — APPROVED** [review]
Small comments as per below, but not a blocker.

LGTM :100:

File: connectivity/tests/pod.go (hunk 1 above)
Comment (nebril) [reply connectivity/tests/pod.go:139]: `"-H", "X-Very-Secret-Token: 42"` are two args I think :thinking:

Comment (sayboras) [reply connectivity/tests/pod.go:139]: and that is why there are three types of people in the world, those can count, and those can't :sweat:

**Pull request merged**